### PR TITLE
Give namepick upon monkey transformation

### PR DESCRIPTION
**Pull request by brndd**
[gameplay]

[I promised I'd do this](https://github.com/vgstation-coders/vgstation13/pull/22693#issuecomment-488072992), so here it is.

:cl:
 * rscadd: monkey transformation now gives you the opportunity to name your new monkey self.

### CODE REVIEW

**Comment by boy2mantwicethefam** [comment]
My powerstrat will become far stronger

File: code/modules/mob/transform_procs.dm [hunk 1]
```diff
@@ -31,6 +31,19 @@
 		return
 	if(mind)
 		mind.transfer_to(new_mob)
+		//namepick
+		spawn(10)
+			var/newname
+			for(var/i = 1 to 3)
+				newname = trimcenter(trim(stripped_input(new_mob,"You have been turned into a monkey! Pick a monkey name for your new monkey self.", "Name change [4-i] [0-i != 1 ? "tries":"try"] left",""),1,MAX_NAME_LEN))
```
Comment (Exxion) [inline code/modules/mob/transform_procs.dm]: I'd say `reject_bad_name()` but that doesn't let you name yourself "monkey"
Still worth considering though

Comment (brndd) [reply code/modules/mob/transform_procs.dm]: Good idea. If they want to name themselves "monkey" they can use the default name.

Comment (Exxion) [reply code/modules/mob/transform_procs.dm]: I believe `reject_bad_name()` includes the trimming, so you can leave out the `trim()` and `trimcenter()`
I'd test it to make sure though

Comment (brndd) [reply code/modules/mob/transform_procs.dm]: I looked at the code and it didn't seem to trim, just reject names that included multiple spaces etc. I think autotrimming is better, but maybe I'm nannying the players.

Comment (Exxion) [reply code/modules/mob/transform_procs.dm]: It claims to trim but looks like it does so with a weird method
Try it

Comment (brndd) [reply code/modules/mob/transform_procs.dm]: Looks like it does work indeed.

**Pull request merged**